gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/BundleFilter.java

```diff
@@ -53,4 +54,20 @@ public Bundle filter(String folderPath, FilterConfiguration filterConfiguration,
         return filteredBundle;
     }
 
+    public void validateImportedFolder(String folderPath, Bundle bundle) {
+        Boolean isValidImport = false;
+
+        // Verifies if node folder is found or if Root Node is found only if the base directory is specified
+        for (Folder folder : bundle.getFolders().values()) {
+            if (folderPath.endsWith(folder.getName()) || 
+                (folderPath.equals("/") && folder.getName().equals("Root Node"))) {
+                isValidImport = true;
+            }
+        }
+
+        if (!isValidImport) {
+            throw new RuntimeException("Unable to find specified folder \"" + folderPath + "\" in the Gateway connection folder path");
```

Comment:
I feel like I have the wrong exception here.

Reply:
Could use "EntityFilterException" here since you've put this method in the BundleFilter